### PR TITLE
Added dropout1d functions to torch frontends (#17808)

### LINKED ISSUE #17808: dropout1d

### PR DESCRIPTION
**Pull request by yunsemr**
I added dropout1d function to torch frontends and a test function for it, then I added dropout1d function to paddle backend to handle the test. 
Close #17808

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by yunsemr** [comment]
I have just edited the description.

**Comment by Sam-Armstrong** [comment]
Thanks for doing that @yunsemr! I think it would actually make more sense to split this into two pull requests - one for the torch frontend and one for the paddle backend. You can keep this one open, just removing the paddle backend changes, then make a new PR for the paddle changes.

**Comment by yunsemr** [comment]
How can I split them, should I close this pull request and create new ones? And one more thing, the backend issue is not open, I wrote it because frontend test failed on paddle backend, so is it a problem that doing something which is not an open issue?

**Comment by yunsemr** [comment]
Actually I am stuck, how can I restart the process because everything I try causes another problem :) I could not understand how to do what you said.

**Comment by Sam-Armstrong** [comment]
@yunsemr You can make a new commit simply changing the paddle backend to what it was before you made any changes. Then push that commit here.

**Comment by yunsemr** [comment]
@Sam-Armstrong I deleted the changes and committed. What then?

**Comment by Sam-Armstrong** [comment]
Great! Now you can open a separate pull request with those changes by creating a new branch from your master, and committing your changes there. Then you can push those changes to a new branch of your fork and make a new PR from there :)

**Comment by yunsemr** [comment]
You mean, I should push both backend and frontend changes to this new branch right? @Sam-Armstrong

**Comment by Sam-Armstrong** [comment]
@yunsemr No just the changes you made to paddle which you removed here. That way we separate it out into two pull requests.

**Comment by Sam-Armstrong** [comment]
Looks good, thanks 🙂

File: ivy/functional/frontends/torch/nn/functional/dropout_functions.py [hunk 1]
```diff
@@ -9,3 +9,9 @@
 @with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
 def dropout(input, p=0.5, training=True, inplace=False):
     return ivy.dropout(input, p, training=training)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+def dropout1d(input, p=0.5, training=True, inplace=False):
+    return ivy.dropout1d(input, p, training=training)
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/torch/nn/functional/dropout_functions.py:17]:
```suggestion
    if inplace:
        return ivy.dropout1d(input, p, training=training, out=input)
    return ivy.dropout1d(input, p, training=training)
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@yunsemr This looks good! Just one change: we need to be able to handle inplace dropout, which we can do by adding the two lines in my suggested change. One we've done that we should be able to merge. Thanks!

**Comment by yunsemr** [comment]
@Sam-Armstrong, I changed the function as you want to. And one more question, I am an applicant so this is the accepted PR to share url in the questionary right?

**Review by Sam-Armstrong — APPROVED** [review]
@yunsemr Yes, you can you this in your application process 😊

I just realised there was a bug here where we were defaulting to using the wrong data format for torch (channels last rather than channels first). I've pushed the change to fix this. Will get this merged now. Great work!

**Comment by yunsemr** [comment]
@Sam-Armstrong Thanks!

**Pull request merged**